include/dlaf/common/index2d.h

```diff
@@ -168,15 +178,49 @@ class Index2D : public internal::basic_coords<IndexT> {
   }
 };
 
-/// Compute coords of the @p index -th cell in a grid with @p ordering and sizes @p dims
+/// Compute Index2D of a linear index inside a 2D grid with specified size and ordering
 /// @param ordering specify linear index layout in the grid
 /// @param dims with number of rows at @p dims[0] and number of columns at @p dims[1]
 /// @param index is the linear index of the cell with specified @p ordering
 template <class Index2DType, typename LinearIndexT>
 Index2DType computeCoords(Ordering ordering, LinearIndexT index,
-                          const std::array<typename Index2DType::IndexType, 2>& dims);
-
+                          const std::array<typename Index2DType::IndexType, 2>& dims) {
+  using IndexType = typename Index2DType::IndexType;
+
+  switch (ordering) {
+    case Ordering::RowMajor:
+      return {static_cast<IndexType>(index / dims[1]), static_cast<IndexType>(index % dims[1])};
+    case Ordering::ColumnMajor:
+      return {static_cast<IndexType>(index % dims[0]), static_cast<IndexType>(index / dims[0])};
+    default:
+      return {};
+  }
 }
+
+/// Compute linear index of an Index2D
```

Comment:
preconditions are missing.

Reply:
Good point. Since I just moved the code to the header, and this method is going to be changed in the PR #190 , do you mind if we move this task to that PR?

Reply:
Good point. Since I just moved the code to the header, and this method is going to be changed in the PR #190 , do you mind if we move this task to that PR?